src/Data/Key.hs

```diff
@@ -59,68 +63,110 @@ module Data.Key (
   , mapWithKeyDefault -- :: TraversableWithKey t => (Key t -> a -> b) -> t a -> t b
   , foldMapWithKeyDefault -- :: (TraversableWithKey t, Monoid m) => (Key t -> a -> m) -> t a -> m
 
+#ifdef MIN_VERSION_semigroupoids
   -- * TraversableWithKey1
   , TraversableWithKey1(..)
   , foldMapWithKey1Default -- :: (TraversableWithKey1 t, Semigroup m) => (Key t -> a -> m) -> t a -> m
+#endif
   ) where
 
 import Control.Applicative
-import Control.Comonad.Trans.Traced
-import Control.Monad.Free
-import Control.Comonad.Cofree
 import Control.Monad.Trans.Identity
 import Control.Monad.Trans.Reader
 import qualified Data.Array as Array
 import Data.Array (Array)
+#if __GLASGOW_HASKELL__ > 710
+import Data.Foldable1 (Foldable1(..))
+#elif defined(MIN_VERSION_semigroupoids)
+import Data.Semigroup.Foldable (Foldable1(..))
+#endif
```

Comment:
Speaking of which, why _does_ this code need to have CPP that's guarded against GHC 8.0? At a glance, I'm not sure what `Foldable1` has anything to do with GHC 8.0.